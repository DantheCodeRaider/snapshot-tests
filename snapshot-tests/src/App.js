import React, { useEffect, useState } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';
import GitHubCard from './components/GitHubCard';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';

function App() {
  const [gitHubName, setGitHubName] = useState('')
  const [gitHubUser, setGitHubUser] = useState('')
  const [gitHubURL, setGitHubURL] = useState('')
  const [gitHubBlog, setGitHubBlog] = useState('')
  const [gitHubImageURL, setGitHubImageURL] = useState('')

  useEffect(() => {
   fetch('https://api.github.com/users/DantheCodeRaider')
      .then(res => res.json())
      .then(data => {
        setGitHubName(data.name)
        setGitHubUser(data.login)
        setGitHubURL(data.html_url)
        setGitHubBlog(data.blog)
        setGitHubImageURL(data.avatar_url)
      })
  }, [])

  return (
    <div className="App">
      <div>
        <h1>Github Profile Info:</h1>
      </div>
      <Row>
        <Col style={{ display: 'flex', justifyContent: 'center' }}>
          <GitHubCard gitHubImageURL={gitHubImageURL} gitHubUser={gitHubUser} gitHubName={gitHubName} gitHubURL={gitHubURL} gitHubBlog={gitHubBlog}/>
        </Col>
      </Row>
    </div>
  );
}

export default App

//curl https://api.github.com/users/DantheCodeRaider
