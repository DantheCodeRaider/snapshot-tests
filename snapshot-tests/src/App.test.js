import { render, screen, waitFor } from '@testing-library/react'
import renderer from 'react-test-renderer'
import App from './App'

const testData = [{
  "login": "DantheCodeRaider",
  "id": 103788290,
  "node_id": "U_kgDOBi-vAg",
  "avatar_url": "https://avatars.githubusercontent.com/u/103788290?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/DantheCodeRaider",
  "html_url": "https://github.com/DantheCodeRaider",
  "followers_url": "https://api.github.com/users/DantheCodeRaider/followers",
  "following_url": "https://api.github.com/users/DantheCodeRaider/following{/other_user}",
  "gists_url": "https://api.github.com/users/DantheCodeRaider/gists{/gist_id}",
  "starred_url": "https://api.github.com/users/DantheCodeRaider/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/DantheCodeRaider/subscriptions",
  "organizations_url": "https://api.github.com/users/DantheCodeRaider/orgs",
  "repos_url": "https://api.github.com/users/DantheCodeRaider/repos",
  "events_url": "https://api.github.com/users/DantheCodeRaider/events{/privacy}",
  "received_events_url": "https://api.github.com/users/DantheCodeRaider/received_events",
  "type": "User",
  "site_admin": false,
  "name": "Dan",
  "company": null,
  "blog": "Https://coderaider.tech/",
  "location": null,
  "email": null,
  "hireable": null,
  "bio": null,
  "twitter_username": null,
  "public_repos": 67,
  "public_gists": 0,
  "followers": 0,
  "following": 3,
  "created_at": "2022-04-15T22:12:02Z",
  "updated_at": "2022-08-19T00:42:29Z"
}]

    test('renders a snapshot', () => {
      const tree = renderer.create(<App />).toJSON()
      expect(tree).toMatchSnapshot()
    })

     test('renders text DantheCodeRaider', () => {
      // Given
      const updateScreen = jest.fn()
      render(<App updateScreen={updateScreen} />)
      // When
      const nameElement = screen.getAllByRole('link')
      userEvent.type(nameElement) 
      userEvent.onload()
      // Then
      expect(nameElement).toBeInTheDocument('DantheCodeRaider')
    })


