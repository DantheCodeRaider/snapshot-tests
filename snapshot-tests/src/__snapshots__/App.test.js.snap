// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <div>
    <h1>
      Github Profile Info:
    </h1>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <div>
        <div
          className="card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            alt="Github profile image"
            className="card-img-top"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            />
            <p
              className="card-text"
            >
              Check out 
              's GitHub below or his protfolio site at 
              <a
                href=""
              />
            </p>
            <a
              href=""
            >
              <button
                className="btn btn-primary"
                disabled={false}
                type="button"
              >
                Link to GitHub profile
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
